src/main/java/ch/njol/skript/expressions/ExprCraftingSlots.java

```diff
@@ -0,0 +1,171 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.util.Utils;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.Material;
+import org.bukkit.event.Event;
+import org.bukkit.event.inventory.PrepareItemCraftEvent;
+import org.bukkit.inventory.CraftingInventory;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.ItemStack;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@Name("Crafting Inventory Slots")
+@Description({"Represents the slots/items in the crafting inventory such as the result item and matrix of the items."})
+@Examples({"on craft:",
+	"\tif crafting result item is paper:",
+	"\t\tset the crafting matrix to air, air, air, paper, diamond, paper, air, air and air",
+	"on preparing crafting:",
+	"\tset crafting result item to wood"})
+@Since("INSERT VERSION")
+public class ExprCraftingSlots extends SimpleExpression<ItemType> {
+	
+	static {
+		Skript.registerExpression(ExprCraftingSlots.class, ItemType.class, ExpressionType.COMBINED,
+				"[the] crafting [inventory] result (slot|item) [of %inventories%]",
+				"%inventories%'[s] crafting [inventory] result (slot|item)",
+				"[the] crafting [inventory] matrix [(slots|items|shape)] [of %inventories%]",
```

Comment:
I think we could use another term to describe the shape like crafting grid or just grid.

Reply:
I will add `grid` to the pattern but will keep the `matrix` as another option since this is how bukkit named it and just to keep things clear for everyone